plugins/scaffolder/src/api.ts

```diff
@@ -170,14 +173,62 @@ export class ScaffolderClient implements ScaffolderApi {
 
   async getTask(taskId: string): Promise<ScaffolderTask> {
     const baseUrl = await this.discoveryApi.getBaseUrl('scaffolder');
-    const url = `${baseUrl}/v2/tasks/${encodeURIComponent(taskId)}`;
+    const taskUrl = `${baseUrl}/v2/tasks/${encodeURIComponent(taskId)}`;
 
-    const response = await this.fetchApi.fetch(url);
-    if (!response.ok) {
-      throw await ResponseError.fromResponse(response);
+    const taskEventsUrl = `${baseUrl}/v2/tasks/${encodeURIComponent(
+      taskId,
+    )}/events`;
+
+    const taskResponse = await this.fetchApi.fetch(taskUrl);
+    if (!taskResponse.ok) {
+      throw await ResponseError.fromResponse(taskResponse);
     }
 
-    return await response.json();
+    const taskEventsResponse = await this.fetchApi.fetch(taskEventsUrl);
+    if (!taskEventsResponse.ok) {
+      throw await ResponseError.fromResponse(taskEventsResponse);
+    }
+
+    const task = (await taskResponse.json()) as ScaffolderTask;
+    const taskEvents =
+      (await taskEventsResponse.json()) as SerializedTaskEvent[];
+
+    const stepIdToTimestamps = taskEvents
+      .filter(event => event.type === 'log')
+      .reduce((acc, event) => {
+        const stepId = event.body.stepId as string;
+        if (stepId) {
+          acc.set(stepId, [...(acc.get(stepId) ?? []), event.createdAt]);
```

Comment:
This is horribly inefficient, you keep re-spreading and re-constructing this array a huge number of times. Why don't you just keep track of a map from id to `{ min, max }`?